xprocess.py

```diff
@@ -217,19 +217,42 @@ def __init__(self, control_dir, process):
     @property
     @abstractmethod
     def args(self):
-        "The args to start the process."
+        """The args to start the process."""
         raise NotImplementedError
 
     @property
     @abstractmethod
     def pattern(self):
-        "The pattern to match when the process has started."
+        """The pattern to be matched when process is starting."""
         raise NotImplementedError
 
+    def startup_callback(self):
+        """Used to assert process responsiveness after pattern match"""
+        return True
+
+    def wait_callback(self):
+        """Assert that process is ready to answer queries using provided
+        callback funtion. Will raise TimeoutError if self.callback does not
+        return True before self.timeout seconds"""
+        while True:
+            if self.startup_callback():
+                return True
```

Comment:
Perhaps would be good to add a small sleep before the `if datatime.now()...` line?

Reply:
Done.